server/src/main/java/org/opensearch/search/internal/ReaderContext.java

```diff
@@ -105,7 +105,15 @@ public void validate(TransportRequest request) {
         indexShard.getSearchOperationListener().validateReaderContext(this, request);
     }
 
-    private long nowInMillis() {
+    protected AbstractRefCounted getRefCounted() {
+        return refCounted;
+    }
+
+    protected AtomicLong getLastAccessTime() {
+        return lastAccessTime;
+    }
```

Comment:
Why not just return `lastAccessTime.get()` that way its immutable?

Reply:
We use this method to update the last access time
   
`getLastAccessTime().updateAndGet(curr -> Math.max(curr, nowInMillis()));`

Reply:
Seems like a violation of the `GET` API

Reply:
okay agreed, made the change to just update in base class.